src/casp/bq_scripts/anndata_to_avro.py

```diff
@@ -53,39 +65,44 @@ def write_avro(generator, parsed_schema, filename, progress_batch_size=10000):
             writer(out, parsed_schema, records)
 
 
-def dump_core_matrix(raw_x, row_lookup, col_lookup, filename):
+def get_x_matrix(adata: AnnData) -> t.Optional[t.Union[np.ndarray, sparse.spmatrix]]:
     """
-    Write the raw X matrix.
+    If there `raw` is not None use `raw.X` otherwise use just X
+    :param adata: Anndata file to use
     """
-    schema = {
-        "doc": "Raw data indexed by cell and feature id in the CAS BigQuery schema",
-        "namespace": "cas",
-        "name": "CellFeature",
-        "type": "record",
-        "fields": [
-            {"name": "cas_cell_index", "type": "int"},
-            {"name": "cas_feature_index", "type": "int"},
-            {"name": "raw_counts", "type": "int"},
-        ],
-    }
-    parsed_schema = parse_schema(schema)
+    if adata.raw is not None:
+        return adata.raw.X
 
-    def raw_counts_generator():
-        coord = raw_x.tocoo(copy=False)
+    return adata.X
 
-        for row, col, dat in zip(coord.row, coord.col, coord.data):
-            cas_cell_index = row_lookup[row]
-            cas_feature_index = col_lookup[col]
 
-            # Todo -- how can we ensure this is safe/right?
-            v_int = int(dat)
-            yield {
-                "cas_cell_index": cas_cell_index.item(),
-                "cas_feature_index": cas_feature_index.item(),
-                "raw_counts": v_int,
-            }
+def dump_core_matrix(
+    row_array, col_array, data_array, row_offset, cas_cell_index_start, cas_feature_index_start, filename
```

Comment:
Type hints

Reply:
I am going to stop commenting about type hints and missing docstrings hereafter.  We need another PR just for that! :)

Reply:
Absolutely. Within this PR, I just wanted to keep bq_scripts untouched as much as possible.